src/legacy/ui/public/url/url.js

```diff
@@ -18,14 +18,46 @@
  */
 
 import _ from 'lodash';
-import { i18n } from '@kbn/i18n';
+import chrome from '../chrome';
 import { uiModules } from '../modules';
-import { AppStateProvider } from '../state_management/app_state';
+import '../state_management/app_state';
 
-uiModules.get('kibana/url')
-  .service('kbnUrl', function (Private) { return Private(KbnUrlProvider); });
+const location = {
```

Comment:
It might be worth adding a comment here to explain that you're essentially recreating the necessary pieces of angular's `$location` service. I think this would help it future readers of the code understand why these methods are designed the way they are.